draft-irtf-cfrg-vdaf.md

```diff
@@ -1190,6 +1190,206 @@ how the nonces are chosen, but security requires that the nonces be unique. See
+the Aggreagtor's recovered output share.
+
+## Ping-Pong Topology (Only Two Aggregators)
+
+For VDAFs with precisely two Aggregators (i.e., `Vdaf.SHARES == 2`), the
+following "ping pong" communication pattern can be used. It is compatible with
+any request/response transport protocol, such as HTTP.
+
+The first state transition, from `Start` to `Continued` or `Rejected`, is
+induced by the following transition rule. No messages are sent or received
+during this transition.
+
+~~~ transition
+def ping_pong_start(Vdaf,
+                    vdaf_verify_key: bytes[Vdaf.VERIFY_KEY_SIZE],
+                    is_leader: bool,
+                    agg_param: Vdaf.AggParam,
+                    nonce: bytes[Vdaf.NONCE_SIZE],
+                    public_share: bytes,
+                    host_input_share: bytes) -> State:
+    try:
+        host_prep_state = Vdaf.prep_init(
+            vdaf_verify_key,
+            0 if is_leader else 1,
+            agg_param,
+            nonce,
+            public_share,
+            host_input_share,
+        )
+        return Continue(host_prep_state)
+    except:
+        return Rejected()
+~~~
+
+If the state is `Rejected`, then processing halts. Otherwise, if the state is
+`Continued`, then processing continues using the following transition rule until
+a terminal state is reached.
+
+Let us call the initiating party the "Leader" and the responding party the
+"Helper". The high-level idea is that the Leader and Helper will take turns
+running the computation locally until input from their peer is required:
+
+* For a 1-round VDAF (e.g., Prio3 ({{prio3}})), the Leader sends its prep share
+  to the Helper, who computes the prep message locally, computes its output
+  share, then sends the prep message to the Leader. Preparation requires just
+  one round trip between the Leader and the Helper.
+
+* For a 2-round VDAF (e.g., Poplar1 ({{poplar1}})), the Leader sends its
+  first-round prep share to the Helper, who replies with the first-round prep
+  message and its second-round prep share. In the next request, the Leader
+  computes its second-round prep share locally, computes its output share, and
+  sends the second-round prep message to the Helper. Finally, the Helper
+  computes its own output share. Preparation requires two round trips.
+
+* In general, each request includes the Leader's prep share for the previous
+  round and/or the prep message for the current round; correspondingly, each
+  response consists of the prep message for the current round and the Helper's
+  prep share for the next round.
+
+The Aggregators proceed in this ping-ponging fashion until a step of the
+computation fails, indicating the report is invalid and should be rejected, or
+the last round is reached and each Aggregator computes an output share. All
+told there there are `ceil((Vdaf.ROUNDS+1)/2)` requests sent.
+
+The message types are as follows:
+
+* `PingPongMessage(prep_share)` - Sent by the Leader to the Helper in the
```

Comment:
nitpick: could this message be called something like `PingPongInitialize`? `PingPongMessage` is a more generic name than necessary IMO -- all of these message types could equally-accurately be called `PingPongMessage`.

Reply:
Done.